### PR TITLE
4 get banner data from api

### LINKED ISSUE #4: Get banner data from API
Implement API system to get banner data from a server

### PR DESCRIPTION
**Pull request by AustinGarrod**
Resolves #4

### CODE REVIEW

File: src/screens/homeScreen.tsx [hunk 1]
```diff
@@ -1,27 +1,114 @@
...
-import { TableData } from '../components/tableData';
 
+// Functions to handle filtering
+/**
+   * Filters out banners that are disabled from array of banners
+   * @param banners Array of banners to be filtered
+   * @returns filtered list of banners
+   */
+const filterDisabledBanners = (banners: Banner[]): Banner[] => {
+  return [...banners].filter((banner) => {
```
Comment (kittenflapz) [inline src/screens/homeScreen.tsx]: pretty sure you can one liner this:

return [...banners].filter((banner) => banner.enabled === true);

File: src/screens/homeScreen.tsx [hunk 2]
```diff
@@ -1,27 +1,114 @@
...
+}
+
+/**
+ * Filters out banners without locations from array of banners
+ * @param banners Array of banners to be filtered
+ * @returns filtered list of banners
+ */
+const filterUnknownLocationBanners = (banners: Banner[]): Banner[] => {
+  return [...banners].filter((banner) => {
+    return banner.lat !== 0 && banner.long !== 0;
```
Comment (kittenflapz) [inline src/screens/homeScreen.tsx]: as above

File: src/screens/homeScreen.tsx [hunk 3]
```diff
@@ -35,7 +122,8 @@ type State = {
   searchText: string,
   firstNameSortDirection?: "ascending" | "descending" | undefined
   lastNameSortDirection?: "ascending" | "descending" | undefined,
-  branchSortDirection?: "ascending" | "descending" | undefined
+  branchSortDirection?: "ascending" | "descending" | undefined,
```
Comment (kittenflapz) [inline src/screens/homeScreen.tsx]: can we have enums for these instead of using strings?

File: src/screens/homeScreen.tsx [hunk 4]
```diff
@@ -69,101 +150,45 @@ class HomeScreen extends Component<Props, State> {
...
-    return [...banners].filter((banner) => {
-      return banner.enabled === true;
-    });
-  }
-
-  /**
-   * Filters out banners without locations from array of banners
-   * @param banners Array of banners to be filtered
-   * @returns filtered list of banners
+   * Called after component mounts successfuly
```
Comment (kittenflapz) [inline src/screens/homeScreen.tsx]: +l

File: src/screens/homeScreen.tsx [hunk 5]
```diff
@@ -69,101 +150,45 @@ class HomeScreen extends Component<Props, State> {
...
-        return 0;
-      });
-    }
+      this.setState({
+        banners: banners,
+        filteredBanners: banners,
+        isDataLoading: false
+      })
+    })
+    .catch(error => { console.log("Failed to load banners", error) });
```
Comment (kittenflapz) [inline src/screens/homeScreen.tsx:188]: does this need to be handled somewhere else, like showing an error to the user?

**Comment by AustinGarrod** [comment]
Notes have been addressed!

**Review by kittenflapz — APPROVED** [review]

**Pull request merged**